x-pack/plugins/ml/public/application/data_frame_analytics/pages/job_map/components/controls.tsx

```diff
@@ -94,27 +150,91 @@ export const Controls: FC<Props> = ({ analyticsId, modelId, details, getNodeData
     };
   }, [cy, deselect]);
 
+  useEffect(() => {
+    // Update elements and close the flyout
+    if (isModalVisible === false && deleteItem === true) {
+      let destIndexNode;
+      if (deleteTargetIndex === true) {
+        const jobDetails = details[nodeId];
+        const destIndex = jobDetails.dest.index;
+        destIndexNode = `${destIndex}-${JOB_MAP_NODE_TYPES.INDEX}`;
+      }
+      updateElements(nodeId, nodeLabel, destIndexNode);
+      setShowFlyout(false);
+    }
+  }, [isModalVisible, deleteItem]);
+
   if (showFlyout === false) {
     return null;
   }
 
-  const nodeDataButton =
-    analyticsId !== nodeLabel &&
+  const button = (
+    <EuiButton size="s" iconType="arrowDown" iconSide="right" onClick={onActionsButtonClick}>
+      <FormattedMessage
+        id="xpack.ml.dataframe.analyticsMap.flyout.nodeActionsButton"
+        defaultMessage="Node actions"
+      />
+    </EuiButton>
+  );
+
+  const items = [
```

Comment:
I think it's worth memoizing these items

Reply:
I did that initially but the linter complains that the memoization would only be called conditionally - since the items are only used when node type is a trained model.